### PR TITLE
[Prim][PIR] reduce_as op brackward decomp

### PR DESCRIPTION
**Pull request by zeroRains**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

反向拆解reduce_as op

### CODE REVIEW

File: test/deprecated/legacy_test/test_reduce_as_op.py [hunk 1]
```diff
@@ -52,6 +52,8 @@ def setUp(self):
         self.inputs = {'x': self.x, 'target': self.y}
         self.outputs = {'out': self.out}
         self.if_enable_cinn()
+        self.prim_op_type = "comp"
```
Comment (cyber-pioneer) [inline test/deprecated/legacy_test/test_reduce_as_op.py]: “prim"

File: paddle/phi/api/yaml/backward.yaml [hunk 2]
```diff
@@ -1984,6 +1984,7 @@
     param : [x]
   kernel :
     func : reduce_as_grad
+  composite: reduce_as_grad(x, target, out_grad, x_grad)
```
Comment (cyber-pioneer) [inline paddle/phi/api/yaml/backward.yaml]: 现在已经可以自动生成，这个字段不需要手动添加了

File: test/deprecated/legacy_test/test_reduce_as_op.py (hunk 1 above)
Comment (zeroRains) [reply test/deprecated/legacy_test/test_reduce_as_op.py]: done

File: paddle/phi/api/yaml/backward.yaml (hunk 2 above)
Comment (zeroRains) [reply paddle/phi/api/yaml/backward.yaml]: done

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**